File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/alarm/service/AlarmService.java
package cc.nobrain.dev.userserver.domain.alarm.service;

import cc.nobrain.dev.userserver.domain.alarm.service.dto.AlarmDto;

import java.util.List;

public interface AlarmService {
    List<AlarmDto> getMemberAlarmList(Long memberId);

    List<AlarmDto> getMyAlarmList();
}

File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/alarm/service/AlarmService.kt
package cc.nobrain.dev.userserver.domain.alarm.service

import cc.nobrain.dev.userserver.domain.alarm.service.dto.AlarmMessageDto
import cc.nobrain.dev.userserver.domain.alarm.service.dto.AlarmRes

interface AlarmService {
    suspend fun getMemberAlarmList(memberId: Long): List<AlarmMessageDto>

    suspend fun getMyAlarmList(): List<AlarmRes>

    suspend fun sendAlarmToMember(memberId: Long, title: String, content: String, url: String? = null)

    suspend fun readAlarm(alarmId: Long)

    suspend fun readAllAlarm()
}